### PR TITLE
Generalize OAuth approval to clicking+typing

### PR DESCRIPTION
**Pull request by jmandel**

### CODE REVIEW

File: testsuite/oauth/authorize/smart.py [hunk 1]
```diff
@@ -11,5 +11,4 @@ class SmartAuthorizer(base.AbstractAuthorizer):
         authorize_url (string): The vendor's authorize endpoint.
     """
     def __init__(self, config, authorize_url):
-        self.config = config
-        self.authorize_url = authorize_url
+        super().__init__(config, authorize_url)
```
Comment (erikwiffin) [inline testsuite/oauth/authorize/smart.py]: Why does Smart still need an authorizer?

Comment (jmandel) [inline testsuite/oauth/authorize/smart.py]: It doesn't. this file should have been deleted :-)

File: testsuite/oauth/authorize/base.py [hunk 2]
```diff
@@ -128,9 +158,10 @@ def open(self):
         self.browser = self._browser()
 
     def close(self):
-        """ Close the selenium webdriver.
+        """ Close the virtual display.
```
Comment (erikwiffin) [inline testsuite/oauth/authorize/base.py]: Selenium webdriver is still being closed here. Maybe a more generic comment like "close the virtual browser"?

File: testsuite/oauth/authorize/base.py [hunk 3]
```diff
@@ -143,3 +174,5 @@ def find(self, selector):
         """ Shorthand for jQuery style element selecting.
         """
         return self.browser.find_element_by_css_selector(selector)
+
+class Authorizer(AbstractAuthorizer): pass
```
Comment (erikwiffin) [inline testsuite/oauth/authorize/base.py]: It looks like you could just get rid of the Abstract Authorizer, move this module up a level, and have an Authorizer class that everything uses. NoneAuthorizer can inherit a concrete class just as well.

File: testsuite/oauth/authorize/base.py [hunk 4]
```diff
@@ -1,10 +1,13 @@
 """ Authorize the SMART API.
 """
+import time
 from abc import ABCMeta
 from urllib import parse
+from pyvirtualdisplay import Display
```
Comment (erikwiffin) [inline testsuite/oauth/authorize/base.py]: Nitpick, but: I generally try to alphabetize the import statements (so `time` would go last) and they are also organized into groups: stdlib, pip modules, and local modules. So pyvirtualdisplay would go down with selenium.

https://www.python.org/dev/peps/pep-0008/#imports

File: testsuite/oauth/authorize/base.py (hunk 3 above)
Comment (jmandel) [inline testsuite/oauth/authorize/base.py]: Agreed!

File: testsuite/oauth/authorize/base.py (hunk 4 above)
Comment (jmandel) [inline testsuite/oauth/authorize/base.py]: Fair enough -- I will follow all of your nitpicks if you update the lints to fail me for breaking them :-)

**Pull request merged**